docs/advanced-topics/networkobject-parenting.md

```diff
@@ -0,0 +1,173 @@
+/// </summary>
+virtual void OnNetworkObjectParentChanged(NetworkObject parentNetworkObject) { }
+```
+
+You need to consider two main code paths when synchronizing `NetworkObject` parenting:
+
+1. At Object Spawn
+    - Client spawns objects including static scene objects and dynamic spawned objects on join.
+    - Serialize `NetworkObject`s with their payloads (such as `NetworkBehaviour`s etc.)
+    - Write `m_IsReparented` and `m_LatestParent` fields to sync on the client-side
+2. During Gameplay
+    - When a valid `NetworkObject` reparenting happens during networked gameplay on the server-side, it is replicated across the network to the connected clients to sync
+    - Write `m_IsReparented` and `m_LatestParent` fields into a `NetworkBuffer` and send that over to all connected clients with `PARENT_SYNC` message type on `MLAPI_INTERNAL` channel
+
+Transform parent synchronization relies on the initial formation of transforms in the scene hierarchy being identical on all standalone instances.
+
+## NetworkObject Reparenting Rules
+
+A few basic `NetworkObject` reparenting rules are listed below.
+
+:::warning Limiting Non-Networked NetworkObject Transform Parenting
+Rules outlined below are applied and enforced even while not networking (not hosting or connected). Specifically, if you were to try reparenting a `NetworkObject` under a non-`NetworkObject`, that'd be invalid and reverted even though you are not hosting or connected to a server.
+:::
+
+
+### Only A Server (or A Host) Can Reparent
+
+Similar to [Ownership](../netcode-basics/networkobject#ownership), only the server (or host, which is both a server and a client at the same time) can control reparenting of a `NetworkObject` in the network.
+
+Clients however, can send RPCs to server and execute a logic server-side that ultimately makes server to reparent a `NetworkObject`.
+
+### Only Reparenting Under A `NetworkObject` (Or To The Root) Is Valid
+
+A `NetworkObject` can only be reparented under another `NetworkObject` (`GameObject` with `NetworkObject` component attached). Only exception is moving a `NetworkObject` to the root of the scene hierarchy.
+
+This is simply due to the fact that MLAPI would not be able to identify & locate new parent on the remote-side if it was a non-`NetworkObject` parent. Again, except moving it to the root because we could identify no parent (root) scenario without `NetworkObject` identification or scene hierarchy traversal.
+
+### Only Reparenting During Networking Is Valid
+
+A `NetworkObject` can only be reparented while networking, in other terms you can only reparent while listening/running as a server.
+
+If you allowed moves while not networking, you would be desynced immediately upon switching to networking. Also reparenting a `NetworkObject` under a non-`NetworkObject` while not networking would sound valid but that would not be replicable on the remote-side since Netcode does not cover full scene hierarchy synchronization.
```

Comment:
Just a suggestion to use different phrasing instead of using "moves", since it could be confused with positional movements? It's used below as well.

Reply:
+1 here. I'd replace "move" with "reparent" in most if not all cases in the doc.